packages/react-components/react-progress/src/utils/clampValue.ts

```diff
@@ -0,0 +1,15 @@
+export const clampValue = (value?: number, max: number): number | undefined => {
+  const internalValue = value !== undefined && value < 0 ? 0 : value !== undefined && value > max ? max : value;
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (value !== undefined && value < 0) {
+      // eslint-disable-next-line no-console
+      console.error(`The prop 'value' must be greater than or equal to zero. Received value: ${value}`);
+    }
+    if (value && max && value > max) {
+      // eslint-disable-next-line no-console
+      console.error(`The prop 'value' must be less than or equal to 'max'. Received value: ${value}, max: ${max}`);
+    }
+  }
```

Comment:
nit: This could be simplified by an early return if value is undefined:
```ts
if (value === undefined) {
  return value;
}
```

Then you can remove all of the `value !== undefined` checks in the function.